grammar/grammar.ebnf

```diff
@@ -58,14 +121,26 @@ literal_date
     = BTICK, [ 2 *  ], ( [ 2 *  ], ( literal_integer ), "/" ), [ 4 *  ], ( literal_integer ), BTICK;
 
 (* Operators *)
-operator
+op_expr
     = ? Math Operators and Logical Operators ?;
 
+op_dot
+    = '.';
+
+op_assign
+    = '=';
+
+op_arr_index
+    = OSQUARE, literal_integer, {',', literal_integer} CSQUARE;
+
 (* Comments *)
 comment
     = "//", { ANY };
 
 (* Miscellaneous *)
+lvalue
```

Comment:
`lvalue` can be identified at runtime

Reply:
Lvalues still need a grammar rule to identify them, since a valid lvalue like `arr[1]` would not be parsed as a valid identifier. One approach could be to have all identifiers be parsed with the `lvalue` rule and then have them checked at runtime. Please let me know what you think the best way is.

Reply:
Agreed. I'll merge this once the assignment operator has been changed to <-